jobs/tests.py

```diff
@@ -16,77 +14,105 @@ class JobModelTests(TestCase):
 
     def setUp(self):
         """Setting up models with mommy."""
-        self.job = mommy.make(Job, ready_to_publish=True)
-        self.job1 = mommy.make(Job, ready_to_publish=False)
-
-    def test_publish_succesfully(self):
-        """Plain publish method test without any twists."""
-        self.assertFalse(self.job.published_date)
-        self.job.publish()
-        self.assertTrue(self.job.published_date, "Job has no published date.")
-
-    def test_publish_unsuccesfully(self):
-        """Publish doesn't work when ready_to_publish field is set to False."""
-        self.assertFalse(self.job1.published_date)
-        self.job1.publish()
-        self.assertFalse(self.job1.published_date, "Job has published date.")
-
-    def test_publish_more_than_once_and_change_values(self):
-        """Publish once, than change ready_to_publish field value,
-        publish once more."""
-        self.assertFalse(self.job.published_date)
-        self.job.publish()
-        self.job.ready_to_publish = False
-        # Right now if we change ready_to_publish to False,
-        # published_date stays as it was and doesn't become None.
-        self.assertTrue(self.job.published_date)
-        self.job.ready_to_publish = True
-        self.job.publish()
-        self.assertTrue(self.job.published_date, "Job has no published date.")
+        self.custom_date = datetime(2016, 2, 14, 15, 30)
+        self.job_not_ready = mommy.make(Job, ready_to_publish=False)
+        self.job_ready_no_exp_date = mommy.make(
+            Job,
+            ready_to_publish=True,
+            expiration_date=None
+        )
+        self.job_ready_custom_exp_date = mommy.make(
+            Job,
+            ready_to_publish=True,
+            expiration_date=self.custom_date
+        )
+
+    def test_publish_without_ready_to_publish(self):
+        """Attempts to publish jobs with ready_to_published=False
+        should results in an assertion error"""
+        self.assertRaises(AssertionError,  self.job_not_ready.publish)
```

Comment:
:+1: If we decide to change the admin action for make published to not set all checked jobs to ready_to_publish, this test should then break (so it's a nice check).  If we do change that admin action, we'll probably want a try/except to catch the assertion error in the admin code.